R/utils-json.R

```diff
@@ -216,19 +216,32 @@ getApprovals <- function(data, chain_nm, legend_nm, appr_var_all, month=NULL, po
     approval_info <- list()
     appr_dates <- NULL
     
-    startTime <- formatDates(data[[chain_nm]]$approvals$startTime, format_str="%Y-%m-%dT%H:%M:%S")
-    endTime <- formatDates(data[[chain_nm]]$approvals$endTime, format_str="%Y-%m-%dT%H:%M:%S")
-    type <- data[[chain_nm]]$approvals$description
-    type <- unlist(lapply(type, function(desc) {
-      switch(desc,
-             "Working" = "appr_working_uv",
-             "In Review" = "appr_inreview_uv",
-             "Approved" = "appr_approved_uv")
-    }))
-    legendnm <- data[[chain_nm]]$approvals$description
-    appr_dates <- data.frame(startTime=startTime, endTime=endTime, type=type, legendnm=legendnm, stringsAsFactors = FALSE)
+    if (!isEmpty(data[[chain_nm]]$approvals$startTime)) {
+      startTime <- flexibleTimeParse(data[[chain_nm]]$approvals$startTime, timezone = data$reportMetadata$timezone)
+      endTime <- flexibleTimeParse(data[[chain_nm]]$approvals$endTime, timezone = data$reportMetadata$timezone)
+      #hacky fix for 9999 year issue which prevents the rectangles from displaying on the graphs 
+      #apologies to the people of 2100 who have to revisit this
+      for (i in 1:length(endTime)) {
+        if (as.Date(endTime)[i] > "2100-12-31") { 
```

Comment:
would it make sense to have a helper function that did this wonky 9999 conversion for corr & this? That way, if we ever find a cleaner way of doing it, we will only have to fix it once?